### ISSUE TITLE
Deduplication

### ISSUE THREAD

**Issue by femesq** [graphql-python/graphene#783]
I was thinking in ways we could make [this #782](https://github.com/graphql-python/graphene/issues/782) work . I've also poked @syrusakbary on [twitter](https://twitter.com/syrusakbary/status/965660876333182976)...

The propose I was thinking about would improve the `middleware` class, adding a function just like Django's [new-style middleware](https://docs.djangoproject.com/en/2.0/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware).

We would have something like this:
```python
class AuthorizationMiddleware(object):
    def resolve(self, next_mw, root, info, **args):
        ## resolver (request) processing here

    def process_response(self, next_mw, response, **args):
        ## result (response) processing here
```

Does it make any sense?

**Comment by jkimbo** [comment]
@femesq you and @keeth made me curious about how you could add deduplication to graphene so I reimplemented graphql-deduplicator in python: https://github.com/graphql-python/graphene/pull/786

It's a pretty simple algorithm and it should work with the corresponding JS inflate method. It doesn't seem to help much with any of our queries at work but hopefully it's useful for someone.

**Comment by jkimbo** [comment]
Closing this since the deduplicator functionality is now available in 2.1.3
